Automattic/sensei
# Fix sensei email generating multiple queries on page load

Resolves https://github.com/Automattic/sensei/issues/6815

## Proposed Changes
* Updated the check before initializing the email so that instead of generating multiple queries before enabling emails, we can do only one which improves the performance. It triggers meta cache queries, but they are very small and fast, unlike the join queries.

## Testing Instructions

1. You should have a way to monitor the queries (Can use the [Query Monitor](https://wordpress.org/plugins/query-monitor/) plugin)
2. Check the generated queries when you reload a page
3. You should see only one sensei_email query logged there

###Before

https://user-images.githubusercontent.com/6820724/232757085-7bf1fbf8-6fb3-4d9c-bf5b-3e509567c5b0.mov

###After
![Screen Shot 2023-04-18 at 3 02 05 PM](https://user-images.githubusercontent.com/6820724/232757155-3e0e5acf-7068-427d-a1ec-1fff85df84de.png)

## Pre-Merge Checklist
